third_party/amd/backend/compiler.py

```diff
@@ -303,16 +303,19 @@ def make_llir(src, metadata, options):
         passes.common.add_canonicalizer(pm)
         passes.common.add_cse(pm)
 
-        if HIPBackend.instrumentation:
-            HIPBackend.instrumentation.patch("llvmir", pm, mod.context)
-
         passes.convert.add_cf_to_llvmir(pm)
         passes.convert.add_arith_to_llvmir(pm)
         passes.common.add_canonicalizer(pm)
         passes.common.add_cse(pm)
         passes.common.add_symbol_dce(pm)
+
         if options.schedule_hint.lower() != "none":
             amd.passes.ttgpuir.lower_instruction_sched_hints(pm, options.arch, options.num_stages)
+
+        # This can not be moved below the di_scope pass
```

Comment:
Why?

Reply:
It generates non-unique debug/line mapping information from those pseudo instructions that causes LLVM to assert.